common/djangoapps/track/shim.py

```diff
@@ -22,9 +22,11 @@ class LegacyFieldMappingProcessor(object):
     def __call__(self, event):
         context = event.get('context', {})
         if 'context' in event:
+            delete_from_context = bool(context.get('event_source', '') != 'task')
```

Comment:
I don't think we need this logic here, because we don't want to selectively preserve context.   For task events, we're artificially creating a context, and so moving stuff from the context to the regular body is an appropriate use of this shim.

Reply:
Removed my changes.